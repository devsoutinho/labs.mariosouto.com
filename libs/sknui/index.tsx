export { Box } from "./src/components/Box/web";
export { Text } from "./src/components/Text/web";