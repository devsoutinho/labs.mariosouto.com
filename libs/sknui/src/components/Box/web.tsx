import React from "react";
import { StyleSheet } from "../../theme/StyleSheet/web";

interface BoxProps {
  tag?: string;
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  // receive anything
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}
export function Box({ tag, styleSheet, children, ...props }: BoxProps) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const Tag = tag || ("div" as any);
  return (
    <Tag
      style={{
        display: "flex",
        flexDirection: "column",
        ...styleSheet,
      }}
      {...props}
    >
      {children}
    </Tag>
  );
}
