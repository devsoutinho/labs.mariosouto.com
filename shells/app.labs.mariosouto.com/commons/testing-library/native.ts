import { render as rtlRender } from "@testing-library/react-native";
export * from "@testing-library/react-native";

interface DefaultWrapperProps {
  children: React.ReactNode;
}
function DefaultWrapper({ children }: DefaultWrapperProps) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return (ui: any, options = {}) => {
    return rtlRender(ui, {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      wrapper: AllTheProviders as any,
      ...options,
    });
  };
};

export const render = () =>
  console.error("Please create your own render method using customRender()");
