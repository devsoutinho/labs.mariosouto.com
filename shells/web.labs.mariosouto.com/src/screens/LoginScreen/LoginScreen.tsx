import React from "react";
import {
  Box,
  Text,
  useTheme,
  ScreenContainer,
  Image,
  Link,
  InputText,
  Button,
  Form,
  useToast,
  Touchable,
} from "@devsoutinho/sknui/web";

export function LoginScreen() {
  const { theme } = useTheme();
  const toast = useToast();

  return (
    <ScreenContainer
      styleSheet={{
        backgroundColor: theme.colors.neutral.x050,
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Box
        styleSheet={{
          width: "100%",
          flex: {
            xs: 1,
            sm: 0,
          },
          maxWidth: {
            sm: "400px",
          },
          paddingTop: theme.spacing.x12,
          paddingBottom: {
            xs: theme.spacing.x_zero,
            sm: theme.spacing.x12,
          },
        }}
      >
        <Box
          styleSheet={{
            display: {
              xs: "flex",
              sm: "none",
            },
            flex: 1,
          }}
        />
        <Box
          styleSheet={{
            alignItems: "center",
          }}
        >
          <Touchable href="/">
            <Image
              src="https://www.datocms-assets.com/85225/1673400260-logo-devsoutinho.png"
              alt="Logo DevSoutinho"
              styleSheet={{
                borderRadius: theme.rounded.full,
                width: theme.spacing.x20,
                height: theme.spacing.x20,
              }}
            />
          </Touchable>
          <Box
            styleSheet={{
              textAlign: "center",
            }}
          >
            <Text
              typographyToken="heading3"
              styleSheet={{
                marginTop: theme.spacing.x6,
              }}
            >
              Entre na sua conta
            </Text>
            <Text
              styleSheet={{
                textAlign: "center",
              }}
            >
              Ou <Link href="/dashboard">entre no modo público</Link>
            </Text>
          </Box>
        </Box>
        <Box
          styleSheet={{
            boxShadow: {
              xs: theme.shadow.default,
            },
            borderRadius: {
              sm: theme.rounded.lg,
            },
            backgroundColor: theme.colors.neutral.x000,
            marginTop: theme.spacing.x6,
            paddingY: theme.spacing.x8,
            paddingX: theme.spacing.x10,
            color: theme.colors.neutral.x500,
          }}
        >
          <Box
            styleSheet={{
              width: "100%",
              margin: "auto",
              maxWidth: {
                xs: "320px",
                sm: "100%",
              },
            }}
          >
            <Form
              onSubmit={() => {
                toast({
                  title: "Falha no login",
                  description: "Conta não cadastrada",
                  status: "error",
                  duration: 9000,
                  isClosable: true,
                });
              }}
            >
              <InputText label="Email" />
              <Button type="submit">Entrar</Button>
            </Form>
            <Box
              styleSheet={{
                marginTop: theme.spacing.x6,
                flexDirection: "row",
                alignItems: "center",
              }}
            >
              <Box
                styleSheet={{
                  flex: 1,
                  height: theme.spacing.x_unit,
                  backgroundColor: theme.colors.neutral.x100,
                }}
              />
              <Text
                styleSheet={{
                  paddingX: theme.spacing.x2,
                }}
              >
                Ou continue com
              </Text>
              <Box
                styleSheet={{
                  flex: 1,
                  height: theme.spacing.x_unit,
                  backgroundColor: theme.colors.neutral.x100,
                }}
              />
            </Box>
            <Box
              styleSheet={{
                flexDirection: "row",
                marginTop: theme.spacing.x6,
              }}
            >
              <Button
                variant="secondary"
                onTap={() => {
                  toast({
                    title: "Falha no login",
                    description: "Esta opção está temporariamente desabilitada",
                    status: "warning",
                    duration: 9000,
                    isClosable: true,
                  });
                }}
                suffix={
                  <Box
                    tag="svg"
                    styleSheet={{
                      width: theme.spacing.x5,
                      height: theme.spacing.x5,
                    }}
                    fill="currentColor"
                    viewBox="0 0 20 20"
                    aria-hidden="true"
                  >
                    <path
                      fill-rule="evenodd"
                      d="M10 0C4.477 0 0 4.484 0 10.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0110 4.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.203 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.942.359.31.678.921.678 1.856 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0020 10.017C20 4.484 15.522 0 10 0z"
                      clip-rule="evenodd"
                    ></path>
                  </Box>
                }
              >
                GitHub
              </Button>
            </Box>
          </Box>
        </Box>
        <Box
          styleSheet={{
            display: {
              xs: "flex",
              sm: "none",
            },
            flex: 1,
            backgroundColor: theme.colors.neutral.x000,
          }}
        />
      </Box>
    </ScreenContainer>
  );
}
