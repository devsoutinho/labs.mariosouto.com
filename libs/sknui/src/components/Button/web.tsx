import React from "react";
import { useTheme } from "../../theme/web";
import { StyleSheet } from "../../theme/StyleSheet/web";
import { Text, Touchable } from "../web";

interface ButtonProps {
  type?: "button" | "submit" | "reset";
  variant: "primary" | "secondary" | "accent" | "ghost" | "destructive";
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
}
export function Button({
  variant,
  styleSheet,
  children,
  prefix,
  suffix,
}: ButtonProps) {
  const { theme } = useTheme();
  const PrefixComponent = prefix;
  const SuffixComponent = suffix;

  const stylesByVariant = {
    primary: {
      color: theme.colors.neutral.x000,
      backgroundColor: theme.colors.primary.x500,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.primary.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    secondary: {
      color: theme.colors.neutral.x500,
      backgroundColor: theme.colors.neutral.x000,
      borderStyle: "solid",
      borderWidth: theme.spacing.x_unit,
      borderColor: theme.colors.neutral.x300,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.neutral.x300} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    accent: {
      color: theme.colors.neutral.x000,
      backgroundColor: theme.colors.accent.x500,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.accent.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    ghost: {
      color: theme.colors.primary.x500,
      backgroundColor: theme.colors.neutral.x000,
      boxShadow: theme.shadow.none,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.primary.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    destructive: {
      color: theme.colors.neutral.x000,
      backgroundColor: theme.colors.negative.x500,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.negative.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
  };

  const currentVariant = stylesByVariant[variant];
  const { backgroundColor } = stylesByVariant[variant];

  const buttonStyles = {
    borderWidth: theme.spacing.x_unit,
    borderColor: "transparent",
    ...currentVariant,
    backgroundColor,
    overflow: "initial",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    flex: 1,
    appearance: "none",
    gap: "8px",
    borderRadius: theme.rounded.md,
    paddingY: theme.spacing.x2,
    paddingX: theme.spacing.x4,
    _placeholder: {
      color: theme.colors.neutral.x400,
    },
    _hover: {
      opacity: 0.8,
    },
    _focusVisible: {
      ...currentVariant?._focusVisible,
      outline: `2px solid transparent`,
      outlineOffset: "2px",
    },
  };

  return (
    <Touchable
      styleSheet={{
        ...styleSheet,
        ...(buttonStyles as unknown as StyleSheet),
      }}
    >
      {PrefixComponent}
      <Text>{children}</Text>
      {SuffixComponent}
    </Touchable>
  );
}

Button.defaultProps = {
  variant: "primary",
};
