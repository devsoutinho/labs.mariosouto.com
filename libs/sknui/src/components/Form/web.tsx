import React from 'react';
import { StyleSheet } from "../../theme/StyleSheet/web";
import { Box } from "../web";

interface FormProps {
  styleSheet?: StyleSheet;
  children: React.ReactNode;
}
export function Form({ styleSheet, children }: FormProps) {
  const tag = "form";

  return <Box tag={tag} styleSheet={{
    ...styleSheet,
  }}>
    {children}
  </Box>;
}
