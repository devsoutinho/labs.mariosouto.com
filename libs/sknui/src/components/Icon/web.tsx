import React from 'react';
import { StyleSheet } from "../../theme/StyleSheet/web";
import { Box } from "../web";

interface IconProps {
  name?: string;
  styleSheet?: StyleSheet;
}
export function Icon({ name, styleSheet }: IconProps) {
  return (
    <Box styleSheet={styleSheet}>
      {name}
    </Box>
  );
}
