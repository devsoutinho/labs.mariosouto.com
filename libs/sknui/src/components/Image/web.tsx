import { Box } from "../Box/web";
import { StyleSheet } from "../../theme/StyleSheet/web";

interface ImageProps {
  src?: string;
  alt: string;
  styleSheet?: StyleSheet;
}
export function Image({ src, alt, styleSheet }: ImageProps) {
  const { objectFit, objectPosition, borderRadius, ...pictureStyleSheet } =
    styleSheet || {};

  return (
    <Box
      tag="picture"
      styleSheet={{
        ...pictureStyleSheet,
        overflow: "hidden",
        borderRadius,
      }}
    >
      <Box
        tag="img"
        src={src}
        alt={alt}
        style={{
          overflow: "hidden",
          borderRadius,
          objectPosition,
          objectFit: objectFit || "cover",
          width: "100%",
          height: "100%",
        }}
      />
    </Box>
  );
}
