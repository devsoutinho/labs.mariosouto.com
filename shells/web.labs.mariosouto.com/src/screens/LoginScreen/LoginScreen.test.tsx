import React from "react";
import { LoginScreen } from "./LoginScreen";
import { customRender } from "@devsoutinho/common-test/testing-library/web";
import { Provider } from "@devsoutinho/sknui/web";

const render = customRender((props) => <Provider {...props} />);

describe("<HomeScreen />", () => {
  it("renders as expected", () => {
    const { container } = render(<LoginScreen />);

    expect(container).toMatchSnapshot();
  });
});
