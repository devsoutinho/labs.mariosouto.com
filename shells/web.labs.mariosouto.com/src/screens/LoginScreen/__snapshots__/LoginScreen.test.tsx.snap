// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> renders as expected 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #F7F7F7;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-top: 48px;
  padding-bottom: 0px;
}

@media screen and (min-width: 640px) {
  .emotion-1 {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
    max-width: 400px;
    padding-bottom: 48px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media screen and (min-width: 640px) {
  .emotion-2 {
    display: none;
  }
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: var(--chakra-colors-transparent);
  border: var(--chakra-borders-none);
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
}

.emotion-4:focus-visible,
.emotion-4[data-focus-visible] {
  border-radius: 2px;
  outline: 2px solid currentColor;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 80px;
  height: 80px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.emotion-8 {
  display: block;
  font-weight: 700;
  line-height: 1;
  font-size: 30px;
  margin-top: 24px;
}

.emotion-9 {
  display: block;
  font-weight: 400;
  line-height: 1.5;
  font-size: 16px;
  text-align: center;
}

.emotion-10 {
  display: inline;
  background: var(--chakra-colors-transparent);
  border: var(--chakra-borders-none);
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10:hover,
.emotion-10[data-hover] {
  opacity: 0.6;
}

.emotion-10:focus-visible,
.emotion-10[data-focus-visible] {
  border-radius: 2px;
  outline: 2px solid currentColor;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1),0 1px 2px -1px rgb(0 0 0 / 0.1);
  background-color: #FFFFFF;
  margin-top: 24px;
  padding-top: 32px;
  padding-bottom: 32px;
  -webkit-padding-start: 40px;
  padding-inline-start: 40px;
  -webkit-padding-end: 40px;
  padding-inline-end: 40px;
  color: #7E7E7E;
}

@media screen and (min-width: 640px) {
  .emotion-11 {
    border-radius: 8px;
  }
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin: auto;
  max-width: 320px;
}

@media screen and (min-width: 640px) {
  .emotion-12 {
    max-width: 100%;
  }
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 40px;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 600;
  line-height: 1.5;
  font-size: 14px;
  color: #515151;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 400;
  line-height: 1.5;
  font-size: 14px;
  width: 100%;
  border-width: 1px;
  border-color: #9E9E9E;
  border-style: solid;
  border-radius: 6px;
  box-shadow: 0 1px 2px 0 rgb(0 0 0 / 0.05);
  -webkit-padding-start: 8px;
  padding-inline-start: 8px;
  -webkit-padding-end: 8px;
  padding-inline-end: 8px;
  padding-top: 12px;
  padding-bottom: 12px;
  outline: 2px solid transparent;
  outline-offset: 2px;
  margin-top: 4px;
}

.emotion-16:focus,
.emotion-16[data-focus] {
  box-shadow: rgb(255, 255, 255) 0px 0px 0px 0px,#F35627 0px 0px 0px 1px,rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
  border-color: #F35627;
}

.emotion-17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: var(--chakra-colors-transparent);
  border: var(--chakra-borders-none);
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  border-width: 1px;
  border-color: var(--chakra-colors-transparent);
  color: #FFFFFF;
  background-color: #F35627;
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1),0 1px 2px -1px rgb(0 0 0 / 0.1);
  overflow: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  gap: 8px;
  border-radius: 6px;
  padding-top: 8px;
  padding-bottom: 8px;
  -webkit-padding-start: 16px;
  padding-inline-start: 16px;
  -webkit-padding-end: 16px;
  padding-inline-end: 16px;
}

.emotion-18:focus-visible,
.emotion-18[data-focus-visible] {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: #FFFFFF 0px 0px 0px 2px,#F35627 0px 0px 0px 4px,rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
}

.emotion-18::-webkit-input-placeholder {
  color: #9E9E9E;
}

.emotion-18::-moz-placeholder {
  color: #9E9E9E;
}

.emotion-18:-ms-input-placeholder {
  color: #9E9E9E;
}

.emotion-18::placeholder {
  color: #9E9E9E;
}

.emotion-18:hover,
.emotion-18[data-hover] {
  opacity: 0.8;
}

.emotion-19 {
  display: block;
  font-weight: 400;
  line-height: 1.5;
  font-size: 16px;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 1px;
  background-color: #E1E1E1;
}

.emotion-22 {
  display: block;
  font-weight: 400;
  line-height: 1.5;
  font-size: 16px;
  -webkit-padding-start: 8px;
  padding-inline-start: 8px;
  -webkit-padding-end: 8px;
  padding-inline-end: 8px;
}

.emotion-24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 24px;
}

.emotion-25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: var(--chakra-colors-transparent);
  border: var(--chakra-borders-none);
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  border-width: 1px;
  border-color: #B1B1B1;
  color: #7E7E7E;
  background-color: #FFFFFF;
  border-style: solid;
  box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1),0 1px 2px -1px rgb(0 0 0 / 0.1);
  overflow: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  gap: 8px;
  border-radius: 6px;
  padding-top: 8px;
  padding-bottom: 8px;
  -webkit-padding-start: 16px;
  padding-inline-start: 16px;
  -webkit-padding-end: 16px;
  padding-inline-end: 16px;
}

.emotion-25:focus-visible,
.emotion-25[data-focus-visible] {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: #FFFFFF 0px 0px 0px 2px,#B1B1B1 0px 0px 0px 4px,rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
}

.emotion-25::-webkit-input-placeholder {
  color: #9E9E9E;
}

.emotion-25::-moz-placeholder {
  color: #9E9E9E;
}

.emotion-25:-ms-input-placeholder {
  color: #9E9E9E;
}

.emotion-25::placeholder {
  color: #9E9E9E;
}

.emotion-25:hover,
.emotion-25[data-hover] {
  opacity: 0.8;
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 20px;
  height: 20px;
  fill: currentColor;
}

.emotion-28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: #FFFFFF;
}

@media screen and (min-width: 640px) {
  .emotion-28 {
    display: none;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <a
          class="emotion-4"
          href="/"
        >
          <picture
            class="emotion-5"
          >
            <img
              alt="Logo DevSoutinho"
              class="emotion-6"
              src="https://www.datocms-assets.com/85225/1673400260-logo-devsoutinho.png"
              style="border-radius: 9999px; object-fit: cover; width: 100%; height: 100%;"
            />
          </picture>
        </a>
        <div
          class="emotion-7"
        >
          <p
            class="emotion-8"
          >
            Entre na sua conta
          </p>
          <p
            class="emotion-9"
          >
            Ou 
            <a
              class="emotion-10"
              href="/dashboard"
            >
              entre no modo público
            </a>
          </p>
        </div>
      </div>
      <div
        class="emotion-11"
      >
        <div
          class="emotion-12"
        >
          <form
            class="emotion-6"
          >
            <div
              class="emotion-14"
            >
              <label
                class="emotion-15"
              >
                Email
                <input
                  aria-describedby="inputtext_undefined_error"
                  aria-errormessage="inputtext_undefined_error"
                  class="emotion-16"
                  id="inputtext_undefined"
                  type="text"
                />
              </label>
              <div
                class="emotion-17"
              />
            </div>
            <button
              class="emotion-18"
              type="submit"
            >
              <p
                class="emotion-19"
              >
                Entrar
              </p>
            </button>
          </form>
          <div
            class="emotion-20"
          >
            <div
              class="emotion-21"
            />
            <p
              class="emotion-22"
            >
              Ou continue com
            </p>
            <div
              class="emotion-21"
            />
          </div>
          <div
            class="emotion-24"
          >
            <button
              class="emotion-25"
            >
              <p
                class="emotion-19"
              >
                GitHub
              </p>
              <svg
                aria-hidden="true"
                class="emotion-27"
                viewBox="0 0 20 20"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 0C4.477 0 0 4.484 0 10.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0110 4.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.203 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.942.359.31.678.921.678 1.856 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0020 10.017C20 4.484 15.522 0 10 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="emotion-28"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
